lib/checks/canAddLikeTermPolynomialNodes.js

```diff
@@ -24,7 +25,7 @@ function canAddLikeTermPolynomialNodes(node) {
   return restTerms.every(term => {
     const haveSameSymbol = sharedSymbol === term.getSymbolName();
     const exponentNode = term.getExponentNode(true);
-    const haveSameExponent = exponentNode.equals(sharedExponentNode);
+    const haveSameExponent = equals(exponentNode, sharedExponentNode);
```

Comment:
Having methods on nodes makes serialization more difficult.  The methods either need to be stripped before calling `JSON.stringify` or they need to the prototype which means that when they're deserialized the method needs to be added to the deserialized version.  Without them cloning becomes trivial as `const clone = JSON.parse(JSON.stringify(ast))`.